edb/server/compiler/compiler.py

```diff
@@ -537,6 +539,73 @@ def _compile_ql_script(
 
         return sql.decode(), argmap
 
+    def _compile_query_args(
+            self, 
+            ctx: CompileContext,
+            subtypes: List[Tuple[str, Any, bool]], 
+    ) -> Tuple[bytes, uuid.UUID]:
+        # TODO(fmoor) perhaps check for zero parameters and 
+        # return a static value (b'...', uuid.UUID(b'...'))
+
+        current_tx = ctx.state.current_tx()
+        schema = current_tx.get_schema(self._std_schema)
+        elements = []
+        for name, schema_type, required in subtypes:
+            # todo fmoor: this doesn't handle qualified type names
+            type_name = schema_type.get_name(schema).get_local_name().name
+            if required:
+                card = qlast.CardinalityModifier.Required
+            else:
+                card = qlast.CardinalityModifier.Optional
+
+            elem = qlast.ShapeElement(
+                expr=qlast.Path(steps=[qlast.Ptr(
+                    ptr=qlast.ObjectRef(name=name),
+                    direction=s_pointers.PointerDirection.Outbound,
+                )]),
+                compexpr=qlast.TypeCast(
+                    cardinality_mod=card,
+                    optional=not required,
+                    expr=qlast.Parameter(name=name),
+                    type=qlast.TypeName(
+                        maintype=qlast.ObjectRef(name=type_name),
+                    ),
+                ),
+                operation=qlast.ShapeOperation(op=qlast.ShapeOp.ASSIGN),
+            )
+
+            elements.append(elem)
+
+        arg_ast = qlast.SelectQuery(
+            orderby=[],
+            result=qlast.Shape(elements=elements),
+        )
+
+        arg_compile_options = qlcompiler.CompilerOptions(
+            modaliases=current_tx.get_modaliases(),
+            implicit_tid_in_shapes=False,
+            implicit_id_in_shapes=False,
+            implicit_tname_in_shapes=False,
+            apply_query_rewrites=False,
+        )
+
+        ir_expr = qlcompiler.compile_ast_to_ir(
```

Comment:
@elprans it seems like compiling an empty shape does not respect `implicit_id_in_shapes=False` and inserts an `id` in the result anyways. Should this behavior change or instead I could namespace arguments and then have `TypeSerializer.describe()` filter based on an argument prefix?

Reply:
This seems to be driven by [this](https://github.com/edgedb/edgedb/blob/929cf4134988cc4e578c79f82d86f24e3bda47af/edb/edgeql/compiler/viewgen.py#L1219).  @msullivan will know more, but it seems to be a safer bet to specialize `describe()` and filter `id` out.